xbmc/music/tags/test/TestTagLoaderTagLib.cpp

```diff
@@ -20,6 +20,155 @@
+#include <taglib/xiphcomment.h>
+#include <taglib/id3v1genres.h>
+
+using namespace TagLib;
+using namespace MUSIC_INFO;
+
+template <typename T>
+class TestTagParser : public ::testing::Test, public CTagLoaderTagLib {
+   public:
+     T value_;
+
+     virtual void SetUp() {
+       // Configure a basic tag..
+       value_.setTitle ("title");
+       value_.setArtist ("artist");
+       value_.setAlbum ("album");
+       value_.setComment("comment");
+       value_.setGenre("Jazz");
+       value_.setYear (1985);
+       value_.setTrack (2);
+     }
+};
+
+
+typedef ::testing::Types<ID3v2::Tag, ID3v1::Tag, ASF::Tag, APE::Tag, Ogg::XiphComment, MP4::Tag> TagTypes;
+TYPED_TEST_CASE(TestTagParser, TagTypes);
+
+TYPED_TEST(TestTagParser, ParsesBasicTag) {
+  // Create a basic tag
+  TypeParam *tg  = &this->value_;
+  CMusicInfoTag tag;
+  EXPECT_TRUE(CTagLoaderTagLib::ParseTag<TypeParam>(tg, NULL, tag));
+
+  EXPECT_EQ(1985, tag.GetYear());
+  EXPECT_EQ(2, tag.GetTrackNumber());
+  EXPECT_EQ(1u, tag.GetArtist().size());
+  if (tag.GetArtist().size() > 0) EXPECT_EQ("artist", tag.GetArtist().front());
+  EXPECT_EQ("album", tag.GetAlbum());
+  EXPECT_EQ("comment", tag.GetComment());
+  EXPECT_EQ(1u, tag.GetGenre().size());
+  if (tag.GetGenre().size() > 0) EXPECT_EQ("Jazz", tag.GetGenre().front());
+  EXPECT_EQ("title", tag.GetTitle());
+}
+
+
+TYPED_TEST(TestTagParser, HandleNullTag) {
+  // A Null tag should not parse, and not break us either
+  CMusicInfoTag tag;
+  EXPECT_TRUE(CTagLoaderTagLib::ParseTag<TypeParam>(NULL, NULL, tag));
+}
+
+template<typename T, size_t N>
+T * end(T (&ra)[N]) {
+      return ra + N;
+}
+
+const char *tags[] = { "APIC", "ASPI", "COMM", "COMR", "ENCR", "EQU2",
+  "ETCO", "GEOB", "GRID", "LINK", "MCDI", "MLLT", "OWNE", "PRIV", "PCNT",
+  "POPM", "POSS", "RBUF", "RVA2", "RVRB", "SEEK", "SIGN", "SYLT",
+  "SYTC", "TALB", "TBPM", "TCOM", "TCON", "TCOP", "TDEN", "TDLY", "TDOR",
+  "TDRC", "TDRL", "TDTG", "TENC", "TEXT", "TFLT", "TIPL", "TIT1", "TIT2",
+  "TIT3", "TKEY", "TLAN", "TLEN", "TMCL", "TMED", "TMOO", "TOAL", "TOFN",
+  "TOLY", "TOPE", "TOWN", "TPE1", "TPE2", "TPE3", "TPE4", "TPOS", "TPRO",
+  "TPUB", "TRCK", "TRSN", "TRSO", "TSOA", "TSOP", "TSOT", "TSRC", "TSSE",
+  "TSST", "TXXX", "UFID", "USER", "USLT", "WCOM", "WCOP", "WOAF", "WOAR",
+  "WOAS", "WORS", "WPAY", "WPUB", "WXXX",  "ARTIST", "ARTISTS",
+  "ALBUMARTIST" , "ALBUM ARTIST", "ALBUMARTISTS" , "ALBUM ARTISTS", "ALBUM",
+  "TITLE", "TRACKNUMBER" "TRACK", "DISCNUMBER" "DISC", "YEAR", "GENRE",
+  "COMMENT", "CUESHEET", "ENCODEDBY", "COMPILATION", "LYRICS",
+  "REPLAYGAIN_TRACK_GAIN", "REPLAYGAIN_ALBUM_GAIN", "REPLAYGAIN_TRACK_PEAK",
+  "REPLAYGAIN_ALBUM_PEAK", "MUSICBRAINZ_ARTISTID",
+  "MUSICBRAINZ_ALBUMARTISTID", "RATING", "MUSICBRAINZ_ALBUMARTIST",
+  "MUSICBRAINZ_ALBUMID", "MUSICBRAINZ_TRACKID", "METADATA_BLOCK_PICTURE",
+  "COVERART"
+};
+
+
+// This test exposes a bug in taglib library (#671) so for now we will not run it for all tag types
```

Comment:
Thank you for the work.
I can't find #671 can you point me to it or did you mean #670?